exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt

```diff
@@ -199,6 +199,15 @@ object SchemaUtils {
                         else -> processForDefaultValue(exp)
                     }
 
+                    is Float, is Double -> value.toString().let {
+                        when {
+                            // MySQL returns floating-point numbers without a decimal part as integer strings, whereas other databases append a trailing zero.
+                            // For example, the value 30f would be `"30"` in MySQL but `"30.0"` in other databases.
+                            currentDialect is MysqlDialect && it.endsWith(".0") -> it.replace(".0", "")
+                            else -> it
+                        }
+                    }
+
```

Comment:
How about we override the function `nonNullValueAsDefaultString` in `FloatColumnType` and `DoubleColumnType` and place this logic there? The function `nonNullValueAsDefaultString` was created for this purpose.

```
override fun nonNullValueAsDefaultString(value: Float): String = value.toString().let {
    when {
        // MySQL returns floating-point numbers without a decimal part as integer strings, whereas other databases
        // append a trailing zero.
        // For example, the value 30f would be `"30"` in MySQL but `"30.0"` in other databases.
        currentDialect is MysqlDialect && it.endsWith(".0") -> it.replace(".0", "")
        else -> it
    }
}
```

Reply:
Thank you, I didn't know about the method `nonNullValueAsDefaultString()`, and already had thoughts we should add something like this >_<

I also found that blob and json defaults are also probably broken, and this method will be useful to fix them =)